### PR TITLE
GCW-2780 add UI for weight attribute

### PR DESCRIPTION
**Pull request by abhinav098**
Hello Team,

Ticket Link: 
https://jira.crossroads.org.hk/browse/GCW-2780

ScreenShots: Pasted in the ticket. Please take a look there.

This PR adds the UI for weight columns on the form and on item details screen with inline edit.
Please review.
Thanks

### CODE REVIEW

File: app/controllers/items/new.js [hunk 1]
```diff
@@ -169,6 +170,7 @@ export default Ember.Controller.extend({
       length: _this.get("length"),
```
Comment (patrixr) [inline app/controllers/items/new.js]: If anyone gets a change, we should remove all those useless `_this` references :)
we can use lambdas or `bind` to handle scope.

just a side note for future reference 👍

Comment (abulsayyad123) [reply app/controllers/items/new.js]: I've already did this at many places in this file :)

Comment (abhinav098) [reply app/controllers/items/new.js]: Its done already @patrixr .

File: app/templates/items/new.hbs [hunk 2]
```diff
@@ -125,45 +125,87 @@
...
-              {{t "items.new.validation_errors.blank_location"}}
-            {{/if}}
-          </div>
+      <div class="{{if (is-or invalidLocation invalidScanResult) 'ember-view form__control form__control--error'}}"
+        id="inventory_location">
+        <i class="location-icon" aria-hidden="true">{{fa-icon 'map-marker-alt'}}</i>
+        {{input value=locationName disabled=true required='true'}}
+        <div class='input-error'>
+          {{#if invalidScanResult}}
+          {{t "items.new.validation_errors.invalid_location"}}
```
Comment (abulsayyad123) [inline app/templates/items/new.hbs]: Indentation is not proper.

File: app/controllers/items/new.js [hunk 3]
```diff
@@ -100,6 +101,15 @@ export default GoodcityController.extend({
     }
   }),
 
+  showPiecesInput: Ember.computed("codeId", function() {
+    let selected;
+    let codeId = this.get("codeId");
+    if (codeId.length) {
```
Comment (abulsayyad123) [inline app/controllers/items/new.js]: Why `length` check is required ? I guess it is just a number.

File: app/templates/items/new.hbs [hunk 4]
```diff
@@ -194,10 +217,10 @@
 
   {{#if showPublishItemCheckBox}}
     <div class="row">
-      <div class="small-10 columns item-label">
-        <label for="publishItem">{{t "items.new.publish"}}</label>
+      <div class="small-3 columns item-label">
+        <label for="publishItem">Publish</label>
```
Comment (abulsayyad123) [inline app/templates/items/new.hbs]: Missing translation

**Pull request merged**